iogt/middleware.py

```diff
@@ -0,0 +1,62 @@
+from urlparse import urlparse
+
+from django.conf import settings
+from django.http import HttpResponsePermanentRedirect
+
+
+def clean_path(path):
+    '''
+    Converts string of URL paths to list of path elements
+    '''
+    if path == u'/':
+        return [u'/']
+    else:
+        return filter(lambda a: a != '', path.split('/'))
```

Comment:
Nothing _technically_ wrong with paths with multiple `/`. We just want to ignore a first leading or final trailing one. But they're also rare/weird enough to maybe not matter.

`filter` will probably make this Python 3-unfriendly.

URLs are hard I'm sorry :(

Reply:
@JayH5 I can update `filter` to `list(filter(...)` to make it python3 compatible.
But do you want me to revert back to checking paths as strings as opposed to lists, assuming I've checked for trailing and leading slashes?

Reply:
@nathanbegbie list comp is probably best:
```python
return [a for a in path.split(u'/') if a]
```
Still think comparing segments is definitely better than strings.